src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/BuildMacroInfo.cs

```diff
@@ -48,7 +48,7 @@ public int GetBuildMacroValue(string bstrBuildMacroName, out string? pbstrBuildM
             ProjectSystem.Properties.IProjectProperties commonProperties = _configuredProject.Value.Services.ProjectPropertiesProvider.GetCommonProperties();
             pbstrBuildMacroValue = _threadingService?.ExecuteSynchronously(() => commonProperties.GetEvaluatedPropertyValueAsync(bstrBuildMacroName));
 
-            if (pbstrBuildMacroValue == null)
+            if (string.IsNullOrEmpty(pbstrBuildMacroValue))
```

Comment:
Behaviour change. Considers empty values as a failure. This was required to make unit tests pass after I changed the mock to never return null for evaluated values.

Reply:
I think this will break the Build Events property page, see BuildEventsPropPage.GetTokenValue.